tsconfig.json

```diff
@@ -11,7 +11,7 @@
     "noUnusedParameters": true,
     "experimentalDecorators": true,
     "target": "es2015",
-    "lib": ["es2015", "es2016", "es2017", "esnext.asynciterable"]
+    "lib": ["dom", "es2015", "es2016", "es2017", "esnext.asynciterable"]
```

Comment:
`dom` lib is required by `graphql-config` for its internal mechanism for fetching remote schemas.

Reply:
I think you probably want the node lib instead in that case, which I think includes fetch typings? Either that or one of the polyfill libraries' typings should be defining those?

Reply:
there is some weirdness around this one. the node libs are also using the dom lib for these definitions. i need to dig a bit further to make sure my world view is sane.

Reply:
The world view is a bit troubling, [`graphql-config`](https://github.com/prisma/graphql-config/blob/master/tsconfig.json#L8) and [`graphql-request`](https://github.com/prisma/graphql-request/blob/master/tsconfig.json#L11) (dependency of `graphql-config`) are both loading the dom lib. I agree that we should not be doing this (and [so does mhegazy](https://github.com/Microsoft/TypeScript/issues/4948#issuecomment-281513657)). I added the dom lib originally because [someone suggested it](https://github.com/prisma/graphql-request/issues/26#issuecomment-354482903) without really thinking about the consequences. I think we may need to inject ambient types to cover the missing two until `graphql-request` [removes its dependence](https://github.com/prisma/graphql-request/pull/100) on the dom lib. It looks like internally it is using [`cross-fetch/polyfill`](https://github.com/lquixada/cross-fetch) which is using [`node-fetch`](https://github.com/lquixada/cross-fetch/blob/master/dist/node-ponyfill.js#L1) internally. So i'll put some effort into producing some ambient types based on the [`@types/node-fetch`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-fetch/index.d.ts) exports.